examples/theano/robo_cartpole.py

```diff
@@ -0,0 +1,42 @@
+import gym, roboschool
+import numpy as np
+import random
+
+class NaivePolicy:
+    "Simple multi-layer perceptron policy, no internal state"
+    def __init__(self, observation_space, action_space):
+        return
+
+    def act(self, ob):
+        x = np.array([0.5-random.random()])
+        return x
+
+def demo_run():
+    env = gym.make("RoboschoolInvertedPendulum-v1")
+    pi = NaivePolicy(env.observation_space, env.action_space)
+
+    while 1:
+        frame = 0
+        score = 0
+        restart_delay = 0
+        obs = env.reset()
+
+        while 1:
+            a = pi.act(obs)
+            obs, r, done, _ = env.step(a)
+            score += r
+            frame += 1
+            still_open = env.render("human")
+            if still_open==False:
+                return
+            if not done: continue
+            if restart_delay==0:
+                print("score=%0.2f in %i frames" % (score, frame))
+                restart_delay = 60*2  # 2 sec at 60 fps
+            else:
+                restart_delay -= 1
+                if restart_delay==0: break
```

Comment:
restart_delay is initialized as 0, and the score message is printed when restart_delay is zero. But after initialization, once restart_delay is zero again, you break the while loop and the score is restarted. Not really sure what you want to accomplish here.
Usually, you run the training until done is true, and if you don't want to clutter the stdout with score messages, you can define a period of steps to print the score message using the modulo operator.

Reply:
these code are from roboschool example to make a time break between two episodes. i can delete it, but the program will flash very fast.

Reply:
That is desirable if I'm not rendering. I want it to sample as fast as possible. Only the viewer might need to be slowed down.